src/utils/apub_name.rs

```diff
@@ -0,0 +1,96 @@
+use cfg_if::cfg_if;
+use std::sync::LazyLock;
+
+pub fn create_user_apub_name(actor_id: &str) -> String {
+  create_apub_name::<'@'>(actor_id)
+}
+
+pub fn create_community_apub_name(actor_id: &str) -> String {
+  create_apub_name::<'!'>(actor_id)
+}
+
+fn format_apub_name<const PREFIX: char>(instance: &str, name: &str) -> String {
+  format!("{PREFIX}{name}@{instance}")
+}
+
+fn create_apub_name<const PREFIX: char>(actor_id: &str) -> String {
+  const REGEX_STR: &str =
+    r"https?:\/\/((?:(?:[^.]\.)*[^.]+\.[^:]+)|localhost)(?::\d{1,5})?\/[uc]\/([^?#]+)";
+
+  cfg_if! {
+      if #[cfg(feature = "ssr")] {
+        use regex::Regex;
+        static INSTANCE_REGEX: LazyLock<Regex> = LazyLock::new(|| Regex::new(REGEX_STR)
+          .expect("Could not create server side regex.")
+        );
+
+        INSTANCE_REGEX
+          .captures(actor_id)
+          .map_or_else(String::new, |captures| format_apub_name::<PREFIX>(&captures[1], &captures[2]))
+      } else {
+        use js_sys::RegExp;
```

Comment:
Using the browser's regex engine for WASM so we don't need to send a whole regex engine's worth of code in the WASM bundle.